rust/src/database/database.rs

```diff
@@ -67,6 +67,15 @@ impl Database {
         Ok(Self { name, replicas, server_connections })
     }
 
+    #[cfg_attr(feature = "sync", maybe_async::must_be_sync)]
+    pub(super) async fn get(name: String, server_connections: HashMap<Address, ServerConnection>) -> Result<Self> {
```

Comment:
I actually steal the code from 2.x. If a database is not in the cache, but it got created in the background, we need to "get" it instead of creating it.